### PR TITLE
Task1: Generative AI basics

### PR DESCRIPTION
**Pull request by A1iaksei**
Develop a simple application using Semantic Kernel to generate text.

### CODE REVIEW

File: src/main/java/com/epam/training/gen/ai/service/KernelHistoryService.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * <p>
+ * This service provides a method to process user prompts while preserving chat history.
+ * It uses the {@link Kernel} to invoke AI responses based on the user's input and the
+ * previous chat context. The conversation history is updated after each interaction.
+ */
+@Slf4j
+@Service
+public class KernelHistoryService {
+  private final Kernel kernel;
+  private final ChatHistory chatHistory = new ChatHistory();
```
Comment (EugeneRybalko) [inline src/main/java/com/epam/training/gen/ai/service/KernelHistoryService.java:22]: You could use spring bean with "session" scope and proxy mode "target class", so chat history will be in session scope of a specific client
```java
    @Bean
    @Scope(scopeName = "session", proxyMode = ScopedProxyMode.TARGET_CLASS)
    public ChatHistory chatHistory() {
        return new ChatHistory();
    }
```

**Review by EugeneRybalko — APPROVED** [review]

**Comment by EugeneRybalko** [comment]
Great job!

File: src/main/java/com/epam/training/gen/ai/service/KernelHistoryService.java (hunk 1 above)
Comment (A1iaksei) [reply src/main/java/com/epam/training/gen/ai/service/KernelHistoryService.java:22]: I thought about it, but it won't work. As we know REST is stateless and when connecting a REST client to the application we won't have a session.
Anyway, Thank you for your idea!)

Comment (A1iaksei) [reply src/main/java/com/epam/training/gen/ai/service/KernelHistoryService.java:22]: Thank you for your idea.

**Pull request merged**